src/java/main/org/apache/zookeeper/ClientCnxnSocket.java

```diff
@@ -223,4 +231,24 @@ abstract void doTransport(int waitTimeOut, List<Packet> pendingQueue,
      * finally unblock it when finished.
      */
     abstract void sendPacket(Packet p) throws IOException;
+
+    public ZKClientConfig getClientConfig() {
+        return clientConfig;
+    }
+
+    public void setClientConfig(ZKClientConfig clientConfig) {
```

Comment:
I think it'd be fine to pass clientConfig to the constructor of the classes extending this one instead of having this method.